storage/statedb/database.go

```diff
@@ -540,6 +557,22 @@ func (db *Database) DoesExistCachedNode(hash common.Hash) bool {
 	return ok
 }
 
+// DoesExistNodeInPersistent returns if the node exists on the persistent database or its cache.
+func (db *Database) DoesExistNodeInPersistent(hash common.Hash) bool {
+	// Retrieve the node from DB cache if available
+	if enc := db.getCachedNode(hash); enc != nil {
```

Comment:
This is a bit strange since the function name. According to its name, I think it should directly call `db.diskDB.ReadCachedTrieNode`, shouldn't it?

Reply:
Well, without state migration, if the trie is in fastCache, it means the trie is already in level DB. So I added the check routine like this. :)